ECK Resources Helm Chart (#5781)
elastic/cloud-on-k8s: deploy/eck-stack/README.md

# ECK-Stack

ECK Stack is a Helm chart to assist in the deployment of Elastic Stack components, which are
managed by the [ECK Operator](https://www.elastic.co/guide/en/cloud-on-k8s/current/index.html)

## Supported Elastic Stack Resources

The following Elastic Stack resources are currently supported. 

- Elasticsearch
- Kibana

Additional resources will be supported in future releases of this Helm Chart.

## Prerequisites

- Kubernetes 1.20+
- Elastic ECK Operator

## Installing the Chart

### Installing the ECK Operator

Before using this chart, the Elastic ECK Operator is required to be installed within the Kubernetes cluster.

> Full installation instructions can be found within the [our documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-installing-eck.html)

Reviewer: ```suggestion
Full installation instructions can be found within [our documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-installing-eck.html)
```

To install the ECK Operator using Helm.

```sh
# Add the Elastic Helm Repository
helm repo add elastic https://helm.elastic.co && helm repo update
# Install the ECK Operator cluster-wide
helm install elastic-operator elastic/eck-operator -n elastic-system --create-namespace
```

> Additional ECK Operator Helm installation options can be found within the [our documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-install-helm.html)

Reviewer: ```suggestion
Additional ECK Operator Helm installation options can be found within [our documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-install-helm.html)
```

Reviewer: This sentence is almost identical to the one on L25

### Installing the ECK Stack Chart

The following will install the ECK-Stack chart using the default values, which will deploy an Elasticsearch [Quickstart Cluster](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-elasticsearch.html), and a Kibana [Quickstart Instance](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-kibana.html)

```sh
# Add the Elastic Helm Repository
helm repo add elastic https://helm.elastic.co && helm repo update

# Install the ECK-Stack helm chart
# This will setup a 'quickstart' Elasticsearch cluster, and Kibana instance
$ helm install my-release -n my-namespace elastic/eck-stack --create-namespace
```

More information on the different ways to use the ECK Stack chart to deploy Elastic Stack resources
can be found in [our documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/index.html).

## Uninstalling the Chart

To uninstall/delete the `my-release` deployment from the 'my-namespace' namespace:

```console
$ helm delete my-release -n my-namespace
```

The command removes all the Elastic Stack resources associated with the chart and deletes the release.

## Configuration

> The following table lists the configurable parameters of the cert-manager chart and their default values.

Reviewer: ```suggestion
The following table lists the configurable parameters of the eck-stack chart and their default values.
```
